azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -1455,6 +1455,249 @@ def test_no_matching_request_id(self, mocker, stage, pending_op, event, fake_uui
         assert not pending_op.completed
 
 
+@pytest.mark.describe(
+    "CoordinateRequestAndResponseStage - .handle_pipeline_event() -- Called with ConnectedEvent"
+)
+class TestCoordinateRequestAndResponseStageHandlePipelineEventWithConnectedEvent(
+    CoordinateRequestAndResponseStageTestConfig, StageHandlePipelineEventTestBase
+):
+    @pytest.fixture
+    def event(self):
+        return pipeline_events_base.ConnectedEvent()
+
+    def make_new_request_response_op(self, mocker, id):
+        return pipeline_ops_base.RequestAndResponseOperation(
+            request_type="some_request_type",
+            method="SOME_METHOD",
+            resource_location="some/resource/location/{}".format(id),
+            request_body="some_request_body",
+            callback=mocker.MagicMock(),
+        )
+
+    @pytest.fixture
+    def stage(self, mocker, cls_type, init_kwargs):
+        stage = cls_type(**init_kwargs)
+        stage.pipeline_root = pipeline_stages_base.PipelineRootStage(
+            pipeline_configuration=mocker.MagicMock()
+        )
+        stage.send_event_up = mocker.MagicMock()
+        stage.send_op_down = mocker.MagicMock()
+
+        return stage
+
+    @pytest.mark.it(
+        "Sends down a new RequestOperation if the previous RequestOperation never completed"
```

Comment:
it is not a completely brand new request operation right ? it is one of the old ones that were already sent before...so may be `sends down a secondary request operation on being connected if the previous one was pending`...not sure of secondary is the right word here...